iohk-monitoring/src/Cardano/BM/Backend/Log.lhs

```diff
@@ -165,27 +165,33 @@ scribeSettings =
     KC.ScribeSettings bufferSize
 
 registerScribes :: [ScribeDefinition] -> K.LogEnv -> IO K.LogEnv
-registerScribes [] le = return le
-registerScribes (defsc : dscs) le = do
-    let kind      = scKind     defsc
-        sctype    = scFormat   defsc
-        name      = scName     defsc
-        rotParams = scRotation defsc
-        name'     = pack (show kind) <> "::" <> name
-    scr <- createScribe kind sctype name rotParams
-    registerScribes dscs =<< K.registerScribe name' scr scribeSettings le
+registerScribes defscs le =
+    foldM withScribeInEnv le defscs
   where
-    createScribe FileSK ScText name rotParams = mkTextFileScribe
+    withScribeInEnv :: K.LogEnv -> ScribeDefinition -> IO K.LogEnv
+    withScribeInEnv le' defsc = do
+            let kind = scKind defsc
+                sctype = scFormat defsc
+                name = scName defsc
+                rotParams = scRotation defsc
+                name' = pack (show kind) <> "::" <> name
+            scribe <- createScribe kind sctype name rotParams
+            case scribe of
+                Just scr -> K.registerScribe name' scr scribeSettings le'
+                Nothing -> return le'
+
+    createScribe FileSK ScText name rotParams = Just <$> mkTextFileScribe
                                                     rotParams
                                                     (FileDescription $ unpack name)
                                                     False
-    createScribe FileSK ScJson name rotParams = mkJsonFileScribe
+    createScribe FileSK ScJson name rotParams = Just <$> mkJsonFileScribe
                                                     rotParams
                                                     (FileDescription $ unpack name)
                                                     False
-    createScribe StdoutSK sctype _ _ = mkStdoutScribe sctype
-    createScribe StderrSK sctype _ _ = mkStderrScribe sctype
-    createScribe DevNullSK _ _ _ = mkDevNullScribe
+    createScribe StdoutSK sctype _ _ = Just <$> mkStdoutScribe sctype
+    createScribe StderrSK sctype _ _ = Just <$> mkStderrScribe sctype
+    createScribe DevNullSK _ _ _ = Just <$> mkDevNullScribe
+    createScribe JournalSK _ _ _ = return Nothing
```

Comment:
Hm? `Nothing`? What about `mkDevNullScribe`?

Reply:
no, it should not be setup. That's a chicken and egg problem: these scribes are setup from the configuration, but the JournalSK will only be setup after its plugin was loaded. It needs to be handled here to make the case split complete.